server/src/test/java/org/opensearch/search/aggregations/startree/ArrayBasedCollector.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.aggregations.startree;
+
+import org.opensearch.index.compositeindex.datacube.startree.node.StarTreeNode;
+import org.opensearch.search.startree.StarTreeNodeCollector;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ArrayBasedCollector implements StarTreeNodeCollector {
+
+    private final List<StarTreeNode> nodes = new ArrayList<>();
+
+    @Override
+    public void collectStarTreeNode(StarTreeNode node) {
+        nodes.add(node);
+    }
+
+    public boolean matchValues(long[] values) throws IOException {
+        boolean matches = true;
+        for (int i = 0; i < values.length; i++) {
+            matches &= nodes.get(i).getDimensionValue() == values[i];
+        }
+        return matches;
```

Comment:
Can/should you short-circuit this? What if `nodes` has extra values? Also, you can avoid the unnecessary `new long[] { ... }` in the tests by making it a varargs parameter.

```
    public boolean matchValues(long... values) throws IOException {
        for (int i = 0; i < values.length; i++) {
            if (nodes.get(i).getDimensionValue() != values[i]) return false;
        }
        return values.length == nodes.size();
    }
```

Reply:
Changed it to use a Set.
Currently, all tests are matching all collected values. 
So, changed the name to indicate the same.